frame/nfts/src/lib.rs

```diff
@@ -0,0 +1,1725 @@
+		///
+		/// Emits `ItemPriceSet` on success if the price is not `None`.
+		/// Emits `ItemPriceRemoved` on success if the price is `None`.
+		#[pallet::weight(T::WeightInfo::set_price())]
+		pub fn set_price(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			item: T::ItemId,
+			price: Option<ItemPrice<T, I>>,
+			whitelisted_buyer: Option<AccountIdLookupOf<T>>,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let whitelisted_buyer = whitelisted_buyer.map(T::Lookup::lookup).transpose()?;
+			Self::do_set_price(collection, item, origin, price, whitelisted_buyer)
+		}
+
+		/// Allows to buy an item if it's up for sale.
+		///
+		/// Origin must be Signed and must not be the owner of the `item`.
+		///
+		/// - `collection`: The collection of the item.
+		/// - `item`: The item the sender wants to buy.
+		/// - `bid_price`: The price the sender is willing to pay.
+		///
+		/// Emits `ItemBought` on success.
+		#[pallet::weight(T::WeightInfo::buy_item())]
+		pub fn buy_item(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			item: T::ItemId,
+			bid_price: ItemPrice<T, I>,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			Self::do_buy_item(collection, item, origin, bid_price)
+		}
+
+		/// Allows to pay the tips.
+		///
+		/// Origin must be Signed.
+		///
+		/// - `tips`: Tips array.
+		///
+		/// Emits `TipSent` on every tip transfer.
+		#[pallet::weight(T::WeightInfo::pay_tips(tips.len() as u32))]
+		pub fn pay_tips(
+			origin: OriginFor<T>,
+			tips: BoundedVec<ItemTipOf<T, I>, T::MaxTips>,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			Self::do_pay_tips(origin, tips)
+		}
+
+		/// Register a new atomic swap, declaring an intention to send an `item` in exchange for
+		/// `desired_item` from origin to target on the current blockchain.
+		/// The target can execute the swap during the specified `duration` of blocks (if set).
+		/// Additionally, the price could be set for the desired `item`.
+		///
+		/// Origin must be Signed and must be an owner of the `item`.
+		///
+		/// - `collection`: The collection of the item.
+		/// - `item`: The item an owner wants to give.
+		/// - `desired_collection`: The collection of the desired item.
+		/// - `desired_item`: The desired item an owner wants to receive.
+		/// - `maybe_price`: The price an owner is willing to pay or receive for the desired `item`.
+		/// - `maybe_duration`: Optional deadline for the swap. Specified by providing the
+		/// 	number of blocks after which the swap will expire.
+		///
+		/// Emits `SwapCreated` on success.
+		#[pallet::weight(T::WeightInfo::create_swap())]
+		pub fn create_swap(
+			origin: OriginFor<T>,
+			offered_collection: T::CollectionId,
+			offered_item: T::ItemId,
+			desired_collection: T::CollectionId,
+			maybe_desired_item: Option<T::ItemId>,
+			maybe_price: Option<PriceWithDirection<ItemPrice<T, I>>>,
+			duration: <T as SystemConfig>::BlockNumber,
```

Comment:
```suggestion
			maybe_duration: Option<<T as SystemConfig>::BlockNumber>,
```

Reply:
We've agreed to make it non-optional to avoid having perpetual records in storage

Reply:
OK, then docs need to be updated.